docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,283 @@
+>[!NOTE]  
+>As an alternative to the download/upload approach, the authors document the possibility of using provider APIs to perform a server-side object copy. This method could significantly reduce network costs and latency by directly copying the snapshot within the cloud provider's infrastructure. While this option is not implemented in the current version, let's explore its feasibility for adoption in etcd-backup-restore to enable server-side copying in [snapstore](https://github.com/gardener/etcd-backup-restore/blob/master/pkg/types/snapstore.go#L74-L86) during implementation.
+
+##### etcd Controller Enhancements
+
+When a hibernation flow is initiated (by external tooling or higher-level operators), the [etcd controller](https://github.com/gardener/etcd-druid/blob/master/docs/development/controllers.md#etcd-controller) can:
+
+1. Remove etcd’s client ports (2379/2380) from the etcd Service to block application traffic.
+2. Trigger an [on-demand full snapshot](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/05-etcd-operator-tasks.md#trigger-on-demand-fulldelta-snapshot) via an `EtcdOperatorTask`.
+3. Scale down the `StatefulSet` replicas to zero, provided the previous snapshot step is successful.
+4. Create the `ExtendFullSnapshotImmutabilityTask` if `etcd.spec.backup.store.immutability.retentionType` is `"bucket"` and based on `etcd.spec.backup.fullSnapshotSchedule`.
+
+##### Operator Task Enhancements
+
+The `ExtendFullSnapshotImmutabilityTask` will create a cron job that:
+
+- Runs `etcdbrctl extend-snapshot-immutability --gc-from-timestamp=<creation timestamp of task>` to preserve the immutability period of the most recent snapshot. This command re-uploads the latest snapshot, effectively resetting its immutability period. Additionally, it removes any snapshots that have become mutable after the creation timestamp of the task.
+
+By periodically re-uploading (extending) the latest snapshot during hibernation, the authors ensure that the immutability period is extended, and the backups remain **protected throughout the hibernation period**.
+
+###### Lifecycle of `ExtendFullSnapshotImmutabilityTask`
+
+The `ExtendFullSnapshotImmutabilityTask` is **active during hibernation** and is automatically managed by the `etcd-controller`. Its lifecycle is tied to the cluster’s hibernation state:
+
+1. **Task Creation**  
+   - When the etcd cluster enters hibernation (e.g., scaling down to zero replicas), the etcd controller:  
+     - Triggers a final full snapshot.  
+     - Creates the `ExtendFullSnapshotImmutabilityTask` to run `extend-snapshot-immutability --gc-from-timestamp=<creation timestamp of this task>` 
+
+2. **Task Deletion**  
+   - When the cluster resumes from hibernation (scales up to non-zero replicas), the controller:  
```

Comment:
another trigger is when etcd spec is updated and immutability configuration is removed. This is when you will have to also delete this task, right?